backend/lib/fetch/providers.py

```diff
@@ -70,40 +70,39 @@ def _geocode_addresses(addresses, geocoder_name, engine, add_to_db=True):
 
 
 @timed
-def geocode_providers(providers, geocoder_name=GEOCODER, engine=connect.create_db_engine()):
-    """
-    Fetch providers location and IDs from a list of provider inputs.
-
-    Provider inputs must contain (address, npi)
-    and can contain other information (e.g., languages, specialty)
-    """
-    if not providers:
+def geocode_providers(
+    provider_addresses,
+    geocoder_name=GEOCODER,
+    engine=connect.create_db_engine()
+):
+    """Fetch providers locations from list of provider addresses."""
+    if not provider_addresses:
         return []
 
     session = sessionmaker(bind=engine)()
 
     provider_responses = []
 
-    provider_addresses = {provider['address'] for provider in providers}
+    addresses = {address for address in provider_addresses}
     logger.debug('Searching {} addresses for {} providers.'.format(
-        len(provider_addresses), len(providers))
+        len(addresses), len(provider_addresses))
     )
 
     if config.get('address_database'):
         existing_addresses = {
             result.address: {
                 'latitude': result.latitude,
                 'longitude': result.longitude
-            } for result in _fetch_addresses_from_db(provider_addresses, session)
+            } for result in _fetch_addresses_from_db(addresses, session)
         }
         logger.debug('Found {} addresses in DB out of {}.'.format(
-            len(existing_addresses), len(provider_addresses))
+            len(existing_addresses), len(addresses))
         )
     else:
         logger.debug('Address database deactivated.')
         existing_addresses = {}
 
-    addresses_to_geocode = provider_addresses.difference(existing_addresses)
+    addresses_to_geocode = addresses.difference(existing_addresses)
```

Comment:
Technically okay to have addresses be a list if you write it as `existing_addresses.difference(addresses)`